### PR TITLE
restart all C* nodes

### PR DESCRIPTION
**Pull request by snazy**
Also introduced abstraction of cluster jobs.
Now we have two kinds of cluster jobs:
* Cluster job that executes something **on** the executor (repair, cleanup)
* Cluster job that performs something **against** the executor (restart)

### CODE REVIEW

File: cassandra-scheduler/src/main/java/io/mesosphere/mesos/frameworks/cassandra/ApiController.java [hunk 1]
```diff
@@ -574,6 +574,30 @@ public Response updateNodeCount(@QueryParam("nodes") int nodeCount) {
     // repair + cleanup stuff
 
     @GET
```
Comment (BenWhitehead) [inline cassandra-scheduler/src/main/java/io/mesosphere/mesos/frameworks/cassandra/ApiController.java]: `@POST`

File: cassandra-scheduler/src/main/java/io/mesosphere/mesos/frameworks/cassandra/ApiController.java [hunk 2]
```diff
@@ -574,6 +574,30 @@ public Response updateNodeCount(@QueryParam("nodes") int nodeCount) {
...
+    @Path("/cluster/restart/start")
+    public Response clusterRestartStart() {
+        return startJob(CassandraFrameworkProtos.ClusterJobType.RESTART);
+    }
+
+    @GET
```
Comment (BenWhitehead) [inline cassandra-scheduler/src/main/java/io/mesosphere/mesos/frameworks/cassandra/ApiController.java]: `@POST`

File: cassandra-scheduler/src/main/java/io/mesosphere/mesos/frameworks/cassandra/ApiController.java [hunk 3]
```diff
@@ -574,6 +574,30 @@ public Response updateNodeCount(@QueryParam("nodes") int nodeCount) {
...
+    @Path("/cluster/restart/status")
+    public Response clusterRestartStatus() {
+        return jobStatus(CassandraFrameworkProtos.ClusterJobType.RESTART, "clusterRestart");
+    }
+
+    @GET
+    @Path("/cluster/restart/last")
+    public Response lastClusterRestart() {
+        return lastJob(CassandraFrameworkProtos.ClusterJobType.RESTART, "clusterRestart");
+    }
```
Comment (BenWhitehead) [inline cassandra-scheduler/src/main/java/io/mesosphere/mesos/frameworks/cassandra/ApiController.java]: Please add new urls to index document.

**Comment by BenWhitehead** [comment]
Lots of missing `{` `}` on if statements, please add them.

**Comment by snazy** [comment]
Changed these API calls to `@POST`
Updated the `{` `}` things.

**Pull request merged**